pkg/larker/fs/github/github.go

```diff
@@ -0,0 +1,51 @@
+package github
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"github.com/google/go-github/v32/github"
+	"syscall"
+)
+
+var ErrAPI = errors.New("failed to communicate with the GitHub API")
+
+type GitHub struct {
+	client    *github.Client
+	owner     string
+	repo      string
+	reference string
+}
+
+func New(owner, repo, reference, token string) *GitHub {
+	return &GitHub{
+		client:    github.NewClient(nil),
+		owner:     owner,
+		repo:      repo,
+		reference: reference,
+	}
+}
+
+func (gh *GitHub) Get(ctx context.Context, path string) ([]byte, error) {
+	fileContent, _, _, err := gh.client.Repositories.GetContents(ctx, gh.owner, gh.repo, path,
+		&github.RepositoryContentGetOptions{
+			Ref: gh.reference,
+		},
+	)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %v", ErrAPI, err)
```

Comment:
Maybe check if error is about a non-existing file and return `os.ErrNotExist`?

Reply:
See https://github.com/cirruslabs/cirrus-cli/pull/89/commits/aee354c515242d92e1d2e120b008bb7a57987f6a.